src/Semaphore/SemaphoreCharacter.tsx

```diff
@@ -16,7 +17,7 @@ type State = {
 
 class SemaphoreCharacter extends React.Component<Props, State> {
   public static getDerivedStateFromProps(nextProps: Props, prevState: State) {
```

Comment:
Is there a reason you didn't just move this down a level and just forward the character to it directly?

Reply:
Discussed offline.  Also, fixed.

Reply:
I think this function can be removed completely and you can just use `this.props.character` in the render method.